src/terraformCloud/run.ts

```diff
@@ -121,37 +132,46 @@ export const CONFIGURATION_SOURCE: { [id: string]: string } = {
   terraform: 'Terraform',
   'terraform+cloud': 'Terraform Cloud',
   tfeAPI: 'API',
-  tfeModule: 'No-code Module',
+  module: 'No-code Module',
 };
 const cfgSources = Object.keys(CONFIGURATION_SOURCE);
 
+const configurationVersionRelationships = z.object({
+  'ingress-attributes': relationship,
+});
+
 // include=configuration_version (implied from .ingress_attributes too)
 // See https://developer.hashicorp.com/terraform/cloud-docs/api-docs/configuration-versions#show-a-configuration-version
 const configurationVersionAttributes = z.object({
-  source: z.enum([cfgSources[0], ...cfgSources]),
+  source: z.enum([cfgSources[0], ...cfgSources]).nullish(),
```

Comment:
`[cfgSources[0], ...cfgSources]` We duplicate the first entry to end up with an array that contains `ado` twice? What's the intention here?

```suggestion
  source: z.enum(cfgSources).nullish(),
```

Reply:
That decoupling of the first element was intentional but the inclusion of that same element in the 2nd argument was not. This is purely to work around this oddity

![Screenshot 2023-06-13 at 16 04 52](https://github.com/hashicorp/vscode-terraform/assets/287584/ec32482a-a5d3-4c37-8925-3ae5273b09b6)

Reply:
I will fix the exclusion bug.